### DISCUSSION TITLE
Feature Request / Capability Question: Can you extend macOS display arrangement?

### DISCUSSION THREAD

**Discussion by timmystokes** [MonitorControl/MonitorControl#1511] [Ideas]
Hi - glad to have stumbled on this tool, great work!

Myself and many others have been dealing with a display arrangement bug when using multiple of the same displays on macOS. The arrangement configuration seems to frequently and unpredictably swap positions, without any known solutions.

See details here: https://forums.macrumors.com/threads/external-displays-swapping-arrangement-not-remembered.2320211/

I am curious if the display library used here has access to monitor serial numbers, and if so - could these be used to extend the display settings and enhance the arrangement feature such that monitor positioning can be stored via serial ID, allowing them to be distinguished for arrangement purposes?

**Comment by waydabber** [comment]
Hi there, I recommend using [BetterDisplay](https://github.com/waydabber/BetterDisplay#readme) and use the swap identical displays feature.

https://github.com/waydabber/BetterDisplay/issues/1358
(the screenshot in the thread is from an older version but the feature is still there, the app menu looks different).

**Reply by timmystokes** [reply]
What an awesome tool! Thank you for the recommendation, part of me is disappointed to find out conclusively that there is no distinguishability between the displays, but this tool solves the problem so effectively that I'm no longer concerned!

Thank you for your great work!

**Reply by timmystokes** [reply]
I do see that the alphanumeric serial id is unique, I'm curious if it could be used to hardset the arrangement rather than relying on the hotswap whenever it rearranges diplays?

**Reply by waydabber** [reply]
Yes, this is possible. Although macOS can't, the app can identify the display (if set up that way) taking into account the extended identifiers. However layout presets are a future feature so expect this in an update:

https://github.com/waydabber/BetterDisplay/issues/382
